src/libraries/System.Diagnostics.DiagnosticSource/ref/System.Diagnostics.DiagnosticSourceActivity.cs

```diff
@@ -253,7 +253,25 @@ public sealed class ActivityListener : IDisposable
         public System.Diagnostics.SampleActivity<string>? SampleUsingParentId { get { throw null; } set { throw null; } }
         public System.Diagnostics.SampleActivity<ActivityContext>? Sample { get { throw null; } set { throw null; } }
         public void Dispose() { throw null; }
-   }
+    }
+
+    public delegate bool PropagatorGetterCallback(object carrier, string fieldName, out string? value);
+
+    public abstract class TextMapPropagator
+    {
+        public abstract System.Collections.Generic.IEnumerable<string> Fields { get; }
+        public abstract bool Inject(System.Diagnostics.Activity activity, object carrier, Action<object, string, string> setter);
+        public abstract bool Inject(System.Diagnostics.ActivityContext context, object carrier, Action<object, string, string> setter);
+        public abstract bool Inject(System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<string, string?>> baggage, object carrier, Action<object, string, string> setter);
+        public abstract bool Extract(object carrier, string fieldName, PropagatorGetterCallback getter, out string? value);
+        public abstract bool Extract(object carrier, PropagatorGetterCallback getter, out System.Diagnostics.ActivityContext context);
+        public abstract bool Extract(object carrier, PropagatorGetterCallback getter, out System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<string, string?>>? baggage);
```

Comment:
Overriding 6 methods seems like a lot if a simple use case might be

```c#
public sealed class SkipOneActivityLayerPropagator : TextMapPropagator
{
    public override bool Inject(Activity activity, object carrier, Action<object, string, string> setter) =>
        TextMapPropagator.CreateW3CPropagator().Inject(activity.Parent, carrier, setter);
}
```

Maybe this isn't a problem if it won't be common for users to implement these?

Reply:
I am not sure if we need to force the default implementation here as whoever subclass TextMapPropagator should be aware about this implementation to decide if need to override it or not.